docs/cli-local-secrets.md

```diff
@@ -0,0 +1,274 @@
+
+Before diving into the details, make sure you've set up your local environment correctly. If you haven't already, complete the [Quickstart](./cli-quickstart.md) to get Quix CLI up and running. For detailed instructions on setting up your local development environment, refer to our [Local Development](./cli-local-debug.md) tutorial.
+
+## Step 1: Add the Secret to Your `.env` File
+
+Start by adding your secret as an environment variable in the `.env` file. Here’s how to securely store an API token:
+
+```dotenv title=".env" hl_lines="23"
+# ======================================================
+#               Quix CLI guidelines
+# ======================================================
+# To update the quix.yaml from this .env file, use:
+#  quix pipeline update
+#
+# To update the .env file from the quix.yaml file, use:
+#  quix init --update
+# ======================================================
+
+### Quix SDK Configuration ###
+# Configuration settings for QuixStreams
+Quix__Broker__Address=localhost:19092
+
+### Input Topics ###
+# Define the input topics used by the application
+
+### Output Topics ###
+# Define the output topics used by the application
+
+### Secrets ###
+# Sensitive information such as API keys and passwords
+api_secret_token=SECRET-API-TOKEN-VALUE
+
+### Free Text ###
+# Add any free text or comments here
+
+### Untracked Variables ###
+# Variables that are not tracked by Quix CLI
+```
+
+Adding the `api_secret_token` under the "Secrets" section ensures it is handled securely by the Quix CLI.
+
+## Step 2: Update Your Pipeline Configuration
+
+To integrate the new secret into your pipeline, run the following command:
+
+```bash
+$ quix pipeline update
+```
+
+This command updates your pipeline configuration, incorporating the secret into the `quix.yaml` file and generating or updating the `.secrets` file:
+
+```text
+Updating deployments of application starter-source
+✓ starter-source → starter-source 
+✓ 'quix.yaml' was updated
+
+Updating secrets ...
+✓ '.secrets' was updated
+
+✓ The pipeline view was updated
+```
+
+## Step 3: Verify the `quix.yaml` and `.secrets` Files
+
+After running the update command, verify that the `quix.yaml` file now includes the secret key. The key is linked to your secret and referenced by the relevant deployment:
+
+```yaml title="quix.yaml" hl_lines="21"
+# Quix Project Descriptor
+# This file describes the data pipeline and configuration of resources of a Quix Project.
+
+metadata:
+  version: 1.0
+
+# This section describes the Deployments of the data pipeline
+deployments:
+  - name: starter-source
+    application: starter-source
+    version: latest
+    deploymentType: Service
+    resources:
+      cpu: 200
+      memory: 800
+      replicas: 1
+    variables:
+      - name: api_secret_token
+        inputType: Secret
+        required: false
+        secretKey: api_secret_token-key
```

Comment:
Why do we need this -key suffix? the platform doesn't suffix it.

Additionally it won't work either

Reply:
the suffix is there to differentiate: variable name / secret key / secret value, it's not needed but the CLI adds it

I'll replace -key with _key